Add baseline support for RT1160

danieldegrasse opened the pull request:
This PR adds baseline support for the NXP RT1160 platform. It enables the UART on the Cortex M4 core, and the following peripherals on the Cortex M7 core:
- UART
- SPI
- I2C
- CAN
- Watchdog

danieldegrasse committed:
boards: mimxrt1160: Added CAN bus support for RT1160

CAN bus was verified in loopback mode, and transmitted output was
verified using an oscilloscope. Receive functionality was verified in
loopback mode only.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: mimxrt1160: Added SPI support for RT1160

SPI support was verified on lpspi1, by connecting MISO to MOSI

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: mimxrt1160: Enabled ADC support for RT1160

ADC support verified with driver sample. LPADC1 channel 0 pin used,
which maps to pin 10 on J9 on the EVK.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

mbolivar-nordic reviewed (commented):
I won't block this if the NXP maintainers are OK with it, but there are still a few suspicious changes from my perspective.

danieldegrasse committed:
boards: Add support for NXP RT1160 EVK

Add baseline support for mimxrt1160_evk Cortex M4 and M7 cores
UART shell, synchronization, and hello world have all been verified to
build and run correctly.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: mimxrt1160_evk: Fix DTS inconsistencies with reference manual

Fix register addresses in rt11xx SOC dts. Flexcan1 was defined with an
incorrect register address, and watchdog 3 was incorrectly labelled as
watchdog 1. Watchdog 2 should not be enabled, as it is only accessible
from code running in the trustzone context.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: mimxrt1160: Enable watchdog timer on RT1160

Enable watchdog timer on RT1160 evk. This was tested using
the watchdog timer api test (tests/drivers/watchdog/wdt_basic_api)

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse commented:
> 
> 
> @danieldegrasse can you take a look at the CI build failure

@mmahadevan108 Should be fixed now- the watchdog driver sample didn't have a proper definition for the `compatible` string that the RT1160 watchdog uses

dleach02 commented:
@danieldegrasse the PR is showing conflicts with a dtsi file that needs resolving
